src/components/common/YourCompany/StartDate.vue

```diff
@@ -0,0 +1,214 @@
+          </v-btn>
+
+          <!-- Undo changes buttons -->
+          <v-btn v-if="isCorrectionFiling && !onEditMode && isCorrected"
+          text color="primary" id="start-undo-btn" @click="onUndoClicked()">
+            <v-icon small>mdi-undo</v-icon>
+            <span>Undo</span>
+          </v-btn>
+          <v-menu v-if="isCorrectionFiling && !onEditMode && isCorrected"
+            offset-y left nudge-bottom="4" v-model="dropdown">
+            <template v-slot:activator="{ on }">
+              <v-btn text small color="primary" id="start-menu-btn" v-on="on">
+                <v-icon>{{dropdown ? 'mdi-menu-up' : 'mdi-menu-down'}}</v-icon>
+              </v-btn>
+            </template>
+            <v-btn text color="primary" id="more-changes-btn"
+              @click="onChangeClicked(); dropdown = false">
+              <v-icon small color="primary">mdi-pencil</v-icon>
+              <span>{{ editLabel }}</span>
+            </v-btn>
+          </v-menu>
+        </span>
+
+      </v-col>
+    </v-row>
+  </section>
+</template>
+
+<script lang="ts">
+import { Component, Mixins, Prop, Watch } from 'vue-property-decorator'
+import { Getter, Action } from 'vuex-class'
+import { CommonMixin, DateMixin } from '@/mixins/'
+import { DatePicker as DatePickerShared } from '@bcrs-shared-components/date-picker'
+import { ActionBindingIF } from '@/interfaces'
+
+@Component({
+  components: {
+    DatePickerShared
+  }
+})
+export default class StartDate extends Mixins(CommonMixin, DateMixin) {
+  // Global getters
+  @Getter isBenIaCorrectionFiling!: boolean
+  @Getter isFirmCorrectionFiling!: boolean
+  @Getter getCorrectedFilingDate!: string
+  @Getter getBusinessFoundingDate!: string // actually date-time
+  @Getter getCurrentDate!: string
+  @Getter getCurrentJsDate!: Date
+
+  // Global setter
+  @Action setCorrectionFilingDate!: ActionBindingIF
+  @Action setValidComponent!: ActionBindingIF
+
+  /** Whether to show invalid section styling. */
+  @Prop({ default: false })
+  readonly invalidSection: boolean
+
+  protected onEditMode = false
+  protected isCorrected = !!this.getCorrectedFilingDate
+  protected newCorrectedStartDate = undefined // date is "Month Day, Year"
+
+  /** The minimum start date that can be entered (Up to 2 years ago today). */
+  protected get startDateMin (): Date {
+    const startDateMin = new Date(this.getCurrentJsDate)
+    startDateMin.setFullYear(startDateMin.getUTCFullYear() - 2)
+    startDateMin.setHours(0, 0, 0) // Set time to 0 for accurate Date Rules comparison
+
+    return startDateMin
+  }
+
+  /** The minimum start date string. */
+  protected get startDateMinStr (): string {
+    return this.dateToYyyyMmDd(this.startDateMin)
+  }
+
+  /** The maximum start date that can be entered (Up to 90 days from today). */
+  protected get startDateMax (): Date {
+    const startDateMax = new Date(this.getCurrentJsDate)
+    startDateMax.setDate(startDateMax.getUTCDate() + 90)
```

Comment:
This method is really similar to the function in [business-create-ui StartDate](https://github.com/bcgov/business-create-ui/blob/5bb6bd107580b0c0ef10e57c3bf2f2bf0160f347/src/components/Registration/StartDate.vue#L71) however I use getUTCDate instead of getDate because startDateMax is in UTC time and using getDate + 90 sets the max start date the current date plus 2 months in the future so I think  it works on the first of the month but that is it, I verified it in dev on a registration filing.

Reply:
Do we need a new BUG ticket to fix this in Create UI?

Reply:
Yes I think we will, I can make one.

Reply:
What's the bug number?

Reply:
#13127